endpoints/bookstore-grpc/bookstore_server.py

```diff
@@ -0,0 +1,123 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The Python GRPC Bookstore Server Example."""
+
+import argparse
+import time
+
+import bookstore
+import bookstore_pb2
+import status
+
+from google.protobuf import struct_pb2
+from grpc.beta import interfaces
+
+_ONE_DAY_IN_SECONDS = 60 * 60 * 24
+
+
+class BookstoreServicer(bookstore_pb2.BetaBookstoreServicer):
+  """Implements the bookstore API server."""
+  def __init__(self, store):
+    self._store = store
+
+  def ListShelves(self, unused_request, context):
+    with status.context(context):
+      response = bookstore_pb2.ListShelvesResponse()
+      response.shelves.extend(self._store.list_shelves())
+      return response
+
+  def CreateShelf(self, request, context):
+    with status.context(context):
+      (shelf, _) = self._store.create_shelf(request.shelf)
+      return shelf
+
+  def GetShelf(self, request, context):
+    with status.context(context):
+      return self._store.get_shelf(request.shelf)
+
+  def DeleteShelf(self, request, context):
+    with status.context(context):
+      self._store.delete_shelf(request.shelf)
+      return struct_pb2.Value()
+
+  def ListBooks(self, request, context):
+    with status.context(context):
+      response = bookstore_pb2.ListBooksResponse()
+      response.books.extend(self._store.list_books(request.shelf))
+      return response
+
+  def CreateBook(self, request, context):
+    with status.context(context):
+      return self._store.create_book(request.shelf, request.book)
+
+  def GetBook(self, request, context):
+    with status.context(context):
+      return self._store.get_book(request.shelf, request.book)
+
+  def DeleteBook(self, request, context):
+    with status.context(context):
+      self._store.delete_book(request.shelf, request.book)
+      return struct_pb2.Value()
+
+
+def _create_sample_bookstore():
+  """Creates a Bookstore with some initial sample data."""
+  store = bookstore.Bookstore()
+
+  shelf = bookstore_pb2.Shelf()
+  shelf.theme = "Fiction"
+  (_, fiction) = store.create_shelf(shelf)
+
+  book = bookstore_pb2.Book()
+  book.title = "REAMDE"
+  book.author = "Neal Stephenson"
+  store.create_book(fiction, book)
+
+  shelf = bookstore_pb2.Shelf()
+  shelf.theme = "Fantasy"
+  (_, fantasy) = store.create_shelf(shelf)
+
+  book = bookstore_pb2.Book()
+  book.title = "A Game of Thrones"
+  book.author = "George R.R. Martin"
+  store.create_book(fantasy, book)
+
+  return store
+
+
+def _serve():
+  """Configures and runs the bookstore API server."""
+  parser = argparse.ArgumentParser(
```

Comment:
Apply all of my comments about the argparse section in the client to this argparse section as well.